Servers/UI/OJS.Servers.Ui/ClientApp/src/components/contests/contest-password-form/ContestPasswordForm.tsx

```diff
@@ -1,92 +1,65 @@
-import isNil from 'lodash/isNil';
+import { useState } from 'react';
+import { useParams } from 'react-router';
 
-import { useCurrentContest } from '../../../hooks/use-current-contest';
-import { usePageTitles } from '../../../hooks/use-page-titles';
+import useTheme from '../../../hooks/use-theme';
+import { useRegisterUserForContestMutation } from '../../../redux/services/contestsService';
 import Form from '../../guidelines/forms/Form';
-import FormControl, { FormControlType, IFormControlOnChangeValueType } from '../../guidelines/forms/FormControl';
+import FormControl, { FormControlType } from '../../guidelines/forms/FormControl';
 import Heading, { HeadingType } from '../../guidelines/headings/Heading';
 
 import styles from './ContestPasswordForm.module.scss';
 
 interface IContestPasswordFormProps {
-    id: number;
-    isOfficial: boolean;
+  contestName: string;
+  onSuccess: () => void;
+  hasConfirmedParticipation: boolean;
 }
 
-const ContestPasswordForm = ({ id, isOfficial }: IContestPasswordFormProps) => {
-    const {
-        state: {
-            contest,
-            contestPasswordError,
-        },
-        actions: { submitPassword },
-    } = useCurrentContest();
-    const [ passwordValue, setPasswordValue ] = useState<string>('');
+const ContestPasswordForm = (props: IContestPasswordFormProps) => {
+    const { contestName, onSuccess, hasConfirmedParticipation } = props;
 
-    const { actions: { setPageTitle } } = usePageTitles();
+    const { themeColors, getColorClassName } = useTheme();
+    const { contestId: id, participationType } = useParams();
 
-    const passwordFieldName = 'contestPassword';
+    const [ password, setPassword ] = useState<string>('');
+    const [ errorMessage, setErrorMessage ] = useState<string>('');
+    const [ isLoading, setIsLoading ] = useState<boolean>(false);
 
-    const handleOnSubmitPassword = useCallback(async () => {
-        await submitPassword({ id, isOfficial, password: passwordValue });
-    }, [ id, isOfficial, passwordValue, submitPassword ]);
+    const [ registerUserForContest ] = useRegisterUserForContestMutation();
 
-    const handleOnChangeUpdatePassword = useCallback(
-        (value?: IFormControlOnChangeValueType) => {
-            setPasswordValue(isNil(value)
-                ? ''
-                : value.toString());
-        },
-        [ setPasswordValue ],
-    );
-
-    const renderErrorSpan = useCallback(
-        (message: string) => (
-            <span className={styles.errorMessage}>{message}</span>
-        ),
-        [],
-    );
-
-    const renderErrorMessage = useCallback(
-        () => {
-            if (!isNil(contestPasswordError)) {
-                const { detail } = contestPasswordError;
-                return renderErrorSpan(detail);
-            }
+    const textColorClassName = getColorClassName(themeColors.textColor);
+    const isOfficial = participationType === 'compete';
```

Comment:
There is an enum `ContestParticipationType`

Reply:
done